### PR TITLE
5181 integrate e45

### PR DESCRIPTION
**Pull request by weskubo-cgi**
Integrated E45 with Maximus service.
Updated unit tests.
Updated UI to match legacy.

### CODE REVIEW

File: backend/src/main/java/ca/bc/gov/hlth/hnweb/config/WebClientConfig.java [hunk 1]
```diff
@@ -113,6 +128,24 @@ public WebClient hcimWebClient() throws HNWebException {
...
+	    HttpClient httpClient= HttpClient.create().secure(t -> t.sslContext(sslContext));
+		ClientHttpConnector connector= new ReactorClientHttpConnector(httpClient);
+	    
+		return WebClient.builder()
+	    		.clientConnector(connector)
+                .baseUrl(hibcUrl)
+                .filter(logRequest())
+                .defaultHeader(HttpHeaders.CONTENT_TYPE, MediaType.TEXT_PLAIN_VALUE) 
+                .defaultHeader(HttpHeaders.ACCEPT, MediaType.ALL_VALUE) 
+                .defaultHeaders(header -> header.setBasicAuth(hibcUserName, hibcUserPassword))
```
Comment (daveb-hni) [inline backend/src/main/java/ca/bc/gov/hlth/hnweb/config/WebClientConfig.java:145]: Passwords are different for relative url for different HIBC messages, need to move auth header to post call.

**Review by daveb-hni — COMMENTED** [review]
Please see comments.

File: backend/src/main/java/ca/bc/gov/hlth/hnweb/controller/EligibilityController.java [hunk 2]
```diff
@@ -148,166 +127,37 @@
...
+			switch (hwe.getType()) {
+			case DOWNSTREAM_FAILURE:
+				throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR, hwe.getMessage(), hwe);
+			default:
+				throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Bad /lookup-phn request", hwe);				
+			}
+		} catch (WebClientException wce) {
+			throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR, wce.getMessage(), wce);
+		} catch (Exception e) {
+			throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Bad /lookup-phn request", e);
```
[the remark this answers is not in the document]
Comment (weskubo-cgi) [reply backend/src/main/java/ca/bc/gov/hlth/hnweb/controller/EligibilityController.java:159]: Fixed. Error handling will be refactored soon as well.

File: backend/src/test/java/ca/bc/gov/hlth/hnweb/controller/EligibilityControllerTest.java [hunk 3]
```diff
@@ -156,12 +200,150 @@ public void testCheckEligibility_exception() throws ParseException, HL7Exception
 	}
 
 	@Test
-	public void testCheckMspCoverageStatus_success() throws HL7Exception {
+	public void testCheckMspCoverageStatus_success_ineligible() throws HL7Exception, InterruptedException {
```
[the remark this answers is not in the document]
Comment (weskubo-cgi) [reply backend/src/test/java/ca/bc/gov/hlth/hnweb/controller/EligibilityControllerTest.java:203]: Fixed.

File: backend/src/test/java/ca/bc/gov/hlth/hnweb/controller/EligibilityControllerTest.java [hunk 4]
```diff
@@ -156,12 +200,150 @@ public void testCheckEligibility_exception() throws ParseException, HL7Exception
...
+        assertNull(checkCoverageResponse.getCareCardWarning());
+        
+		// Check the client request is sent as expected
+        RecordedRequest recordedRequest = mockBackEnd.takeRequest();        
+        assertEquals(HttpMethod.POST.name(), recordedRequest.getMethod());
+        assertEquals(MediaType.TEXT_PLAIN.toString(), recordedRequest.getHeader(CONTENT_TYPE));		
+	}
+	
+	@Test
+	public void testCheckMspCoverageStatus_error_phnInvalid() throws HL7Exception, InterruptedException {
```
[the remark this answers is not in the document]
Comment (weskubo-cgi) [reply backend/src/test/java/ca/bc/gov/hlth/hnweb/controller/EligibilityControllerTest.java:278]: Fixed.

File: backend/src/main/java/ca/bc/gov/hlth/hnweb/config/WebClientConfig.java (hunk 1 above)
Comment (weskubo-cgi) [reply backend/src/main/java/ca/bc/gov/hlth/hnweb/config/WebClientConfig.java:145]: Fixed. We can roll out this pattern to R15 and R50 as we implement them.

File: backend/src/test/java/ca/bc/gov/hlth/hnweb/controller/EligibilityControllerTest.java [hunk 5]
```diff
@@ -156,12 +200,150 @@ public void testCheckEligibility_exception() throws ParseException, HL7Exception
...
+        assertNull(checkCoverageResponse.getCareCardWarning());
+
+		// Check the client request is sent as expected
+        RecordedRequest recordedRequest = mockBackEnd.takeRequest();        
+        assertEquals(HttpMethod.POST.name(), recordedRequest.getMethod());
+        assertEquals(MediaType.TEXT_PLAIN.toString(), recordedRequest.getHeader(CONTENT_TYPE));		
+	}
+	
+	@Test
+	public void testCheckMspCoverageStatus_error_dateOfBirthMismatch() throws HL7Exception, InterruptedException {
```
[the remark this answers is not in the document]
Comment (weskubo-cgi) [reply backend/src/test/java/ca/bc/gov/hlth/hnweb/controller/EligibilityControllerTest.java]: FIxed.

**Review by daveb-hni — APPROVED** [review]
Looks good

**Pull request merged**